lxd/containers_post.go

```diff
@@ -14,12 +14,57 @@ import (
 	"github.com/lxc/lxd/shared"
 )
 
-func btrfsCopyImage(hash string, name string, d *Daemon) error {
-	dpath := shared.VarPath("lxc", name)
-	imagefile := shared.VarPath("images", hash)
-	subvol := fmt.Sprintf("%s.btrfs", imagefile)
+func btrfsCmdIsInstalled() bool {
+	/*
+	 * Returns true if the "btrfs" tool is in PATH else false.
+	 *
+	 * TODO: Move this to the main code somewhere and call it once?
+	 */
+	out, err := exec.Command("which", "btrfs").Output()
+	if err != nil || len(out) == 0 {
+		return false
+	}
+
+	return true
+}
 
-	return exec.Command("btrfs", "subvolume", "snapshot", subvol, dpath).Run()
+func btrfsIsSubvolume(subvolPath string) bool {
+	/*
+	 * Returns true if the given Path is a btrfs subvolume else false.
+	 */
+	if !btrfsCmdIsInstalled() {
+		return false
+	}
+
+	out, err := exec.Command("btrfs", "subvolume", "show", subvolPath).CombinedOutput()
+	if err != nil || strings.HasPrefix(string(out), "ERROR: ") {
```

Comment:
I've got 3.19 here but that may well be me building the latest upstream in my PPA (ppa:stgraber/stgraber.net) rather than using whatever's in the distro.

Anyway, we can't rely on those return values for now, but maybe we'll be able to in a few years :)